mautic/mautic
# Support a custom country and region list through a local file

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Y 
| Automated tests included? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This supports Mautic hosting providers to support a custom list of countries (and I included regions just in case) without needing a lot of code lift to make it configurable. This requires writing a custom list(s) to the instance's filesystem as configured in the `upload_dir` parameter.

[//]: # ( As applicable: )
#### Steps to test this PR
1. Write the two test files (https://github.com/mautic/mautic/blob/113bfd52d850b997c85b076ca81dca4a52d2e080/app/bundles/LeadBundle/Tests/Helper/files/countries.json, https://github.com/mautic/mautic/blob/113bfd52d850b997c85b076ca81dca4a52d2e080/app/bundles/LeadBundle/Tests/Helper/files/regions.json) included in this PR to the instance's upload dir `media/files` by default
2. Edit a contact and note that country and regions now matches the custom lists
3. Create a new form with a country field and match it to the contact country field. View the form and note that it has the custom field as lists. 

#### Other areas of Mautic that may be affected by the change:
1. Should not affect other areas outside of region and country lists
